contracts/facets/ArchaeologistFacet.sol

```diff
@@ -84,4 +86,80 @@ contract ArchaeologistFacet {
     {
         return s.cursedBonds[archaeologist];
     }
+
+    /// @notice Unwraps the sarcophagus.
+    /// @dev Verifies that the unencrypted shard matches the hashedShard stored
+    /// on chain and pays the archaeologist.
+    /// @param identifier The identifier of the sarcophagus
+    /// @param unencryptedShard The archaeologist's corresponding unencrypted shard
+    /// @return The boolean true if the operation was successful
+    function unwrapSarcophagus(
+        bytes32 identifier,
+        bytes memory unencryptedShard
+    ) external returns (bool) {
+        // Confirm that the sarcophagus exists
+        if (
+            s.sarcophaguses[identifier].state !=
+            LibTypes.SarcophagusState.Exists
+        ) {
+            revert LibErrors.SarcophagusDoesNotExist(identifier);
+        }
+
+        // Confirm that the sender is an archaeologist on this sarcophagus
+        if (!LibUtils.archaeologistExistsOnSarc(identifier, msg.sender)) {
+            revert LibErrors.ArchaeologistNotOnSarcophagus(msg.sender);
+        }
+
+        // Confirm that the resurrection time has passed and that the
+        // resurrection window has not passed
+        LibUtils.unwrapTime(
+            s.sarcophaguses[identifier].resurrectionTime,
+            s.sarcophaguses[identifier].resurrectionWindow
+        );
+
+        // Confirm that the archaeologist has not already unwrapped by checking if the unencryptedShard is empty
+        LibUtils.archaeologistUnwrappedCheck(identifier, msg.sender);
+
+        // Comfirm that the sarcophagus has been finalized
+        if (bytes(s.sarcophaguses[identifier].arweaveTxId).length == 0) {
+            revert LibErrors.SarcophagusNotFinalized(identifier);
+        }
+
+        // Get the archaeologist's data from storage
+        LibTypes.ArchaeologistStorage memory archaeologistData = LibUtils
+            .getArchaeologist(identifier, msg.sender);
+
+        // Confirm that the hash of the unencrypted shard matches the hashedShard in storage
+        if (keccak256(unencryptedShard) != archaeologistData.hashedShard) {
+            revert LibErrors.UnencryptedShardHashMismatch(
+                unencryptedShard,
+                archaeologistData.hashedShard
+            );
+        }
+
+        // Store the unencrypted shard in on the archaeologist object in the sarcophagus
+        s
+        .sarcophagusArchaeologists[identifier][msg.sender]
+            .unencryptedShard = unencryptedShard;
+
+        // Set the sarcophagus state as done
+        s.sarcophaguses[identifier].state = LibTypes.SarcophagusState.Done;
```

Comment:
This should only be set to done if *all* of the archaeologists have completed their job. I would say this could be removed for now, and we should talk through *if* a successful sarcophagus needs to have its state changed (from "Exists")